__tests__/api.test.js

```diff
@@ -128,7 +128,98 @@ describe('GET /api/articles/:article_id', () => {
         });
     })
 })
-
+describe('POST /api/articles/:article_id/comments', () => {
+    test("returns the new comment that has been created along with 201 status", () => {
+        return request(app)
+        .post("/api/articles/2/comments")
+        .send({
+            username: "butter_bridge",
+            body: "well well well"
+        })
+        .expect(201)
+        .then(({body}) => {    
+            const comment = body.comment;
+            expect(comment).toMatchObject({ 
+                body: "well well well",
+                votes: expect.any(Number),
+                article_id: 2,
+                author: "butter_bridge",
+                comment_id:  expect.any(Number)
+            });
+        });
+    })
+    test('if client doesnt supply required fields return status code 400 and a helpful error messge', () =>{
+        return request(app)
+        .post('/api/articles/2/comments')
+        .send({
+        username: "butter_bridge"     
+        })
+        .expect(400)
+        .then(({ body }) => {
+            expect(body.message).toBe('both username and comment body is required')
+        })
+    })
+    test('if client supplies extra fields they are ignored', () => {
+        return request(app)
+        .post("/api/articles/2/comments")
+        .send({
+            username: "butter_bridge",
+            body: "well well well",
+            extrsProp: "extra value"
+        })
+        .expect(201)
+        .then(({body}) => {       
+            const comment = body.comment;
+            expect(comment).toMatchObject({ 
+                body: "well well well",
+                votes: expect.any(Number),
+                article_id: 2,
+                author: "butter_bridge",
+                comment_id:  expect.any(Number)
+            });
+        });
+    }) 
+    test('if user name doesnt exist in database return error code 401 and helpful message', () => {
```

Comment:
we need to change the test description  too

Reply:
done